docs/developer-setup.md

```diff
@@ -0,0 +1,234 @@
+
+### Enable Nextcoud Talk Recording Server
+
+**Important:** Make sure that the HPB is correctly running!
+
+Start the container of the recording server:
+
+```
+docker compose up -d talk-recording
+```
+
+Go to the admin settings of talk and add the recording server (`http://talk-recording.local` with shared secret `6789`)
+
+### Use federation between 2 Nextcloud Servers
+
+The provided `docker-compose.yml` file from `nextcloud-docker-dev` supports spinning up multiple instances out of the box. For the `master/main` branch, up to 3 instances can be started (e.g. `docker compose up -d nextcloud nextcloud2 nextcloud3`). This allows easy testing/development for federated scenarios.
+
+1. Make sure to have 2 instances of Nextcloud running with
+
+    ```
+    docker compose up -d nextcloud nextcloud2
+    ```
+
+2. In case of federated calling, the HPB needs to be enabled on both instances
+
+    ```
+    ./scripts/enable-talk-hpb.sh nextcloud
+    ./scripts/enable-talk-hpb.sh nextcloud2
+    ```
+
+3. Allow self signed certs on both instances
+
+    ```
+    ./scripts/occ.sh nextcloud -- config:system:set sharing.federation.allowSelfSignedCertificates --value true --type bool
+    ./scripts/occ.sh nextcloud2 -- config:system:set sharing.federation.allowSelfSignedCertificates --value true --type bool
+    ```
+
+4. Copy and import the certificates to the other instance
+
+    ```
+    docker compose cp data/ssl/nextcloud2.local.crt nextcloud:/tmp
+    docker compose cp data/ssl/nextcloud.local.crt nextcloud2:/tmp
+
+    ./scripts/occ.sh nextcloud2 -- security:certificates:import /tmp/nextcloud.local.crt
+    ./scripts/occ.sh nextcloud -- security:certificates:import /tmp/nextcloud2.local.crt
+    ```
+
+5. Optional: Verify certs
+
+    ```
+    ./scripts/occ.sh nextcloud2 -- security:certificates
+    ./scripts/occ.sh nextcloud -- security:certificates
+    ```
+
+6. Enable federation in the admin settings of Nextcloud Talk.
+
+### Rebuild / update Talk after code changes
+
+#### JavaScript
+
+If you modify any JavaScript file, either directly or by switching to a different branch, you will need to rebuild them. You can do that with:
+
+```
+cd workspace/server/apps-extra/spreed
+make build-js
+```
+
+For JavaScript development rather than manually rebuilding the files whenever you modify them you can instead run a watcher that will automatically rebuild them when they are changed:
+```
+cd workspace/server/apps-extra/spreed
+make watch-js
+```
+
+Note, however, that in some cases the watcher may stop running (for example, if it gets killed by the system after some time due to its RAM consumption), so you might want to check it from time to time.
+
+No matter if you manually rebuilt the JavaScript files or if you use the watcher note that you will need to force refresh Talk in the browser, as otherwise the previous version of the build JavaScript files could be cached and used instead of the latest one.
```

Comment:
```suggestion
No matter if you manually rebuilt the JavaScript files or if you use the watcher note that you will need to force refresh Talk in the browser (`Fn` + `F5`), as otherwise the previous version of the build JavaScript files could be cached and used instead of the latest one.
```

Reply:
That's dependent on the machine and browser. Not sure we should include it like this.